src/digital.rs

```diff
@@ -2,19 +2,32 @@
 
 /// Single digital output pin
 pub trait OutputPin {
-    /// Is the output pin high?
-    fn is_high(&self) -> bool;
-
-    /// Is the output pin low?
-    fn is_low(&self) -> bool;
-
     /// Sets the pin low
     fn set_low(&mut self);
 
     /// Sets the pin high
     fn set_high(&mut self);
 }
 
+/// Output pin that can read its output state
+#[cfg(feature = "unproven")]
+trait StatefulOutputPin: OutputPin {
+    /// Is the pin set to high?
+    fn is_set_high(&self) -> bool;
+
+    /// Is the pin set to low?
+    fn is_set_low(&self) -> bool;
+
+    /// Toggle pin output
```

Comment:
Sorry if I am late to the party. Can you override a method in a trait? On [some cores](https://github.com/ryankurte/efm32-base/blob/1a8d48edae129075b35cea5ac0a36badebbc0c1d/emlib/inc/em_gpio.h#L925) toggle is exposed as a separate register / no read state is required.

I'd propose having the toggle as a separate trait that you can implement on StatefulOutputPin if toggle function is not available.

Reply:
As discussed in #31 (et al) I also think that different capabilities should be modelled in different composable (and mode dependent) traits instead of being munged together into a few traits that do a lot of things at once. Something like `is_set_low()` on a `StatefulOutputPin` is not clear what it actually means, i.e. is that supposed to be the electrical or the logical state of the pin? For `toggle()` it has to be the latter, not the former, in other cases you might want to have the electrical state, though and cannot have this. Also (as you said) there are implementations allowing an efficient toggle without reading the logical state from a pin. There might even be implementations which do not allow you to read back the logical state, although I don't recall any from the top of my head.

I only agreed to moving it out of `OutputPin` because it belongs there even less and this change turns into into an `unproven` functionality so if it doesn't provide the hoped for benefit (as I suspect it will not) then we can replace it by something better.

Reply:
Sounds reasonable. I'll remove it in favour of adding `ToggleableOutputPin` later.

Reply:
> Can you override a method in a trait?

@ryankurte [Yes, you can.](https://doc.rust-lang.org/book/second-edition/ch10-02-traits.html#default-implementations)